### PR TITLE
Improve CI testing for OpenAPI, new tool `spiceschema`, fix broken OpenAPI stuff.

### PR DESCRIPTION
**Pull request by Jeadie**
# 🗣 Description
**New tool**
```shell
spiceschema http > openapi.json
spiceschema http --yaml 
```

**New CI**
 - `.github/workflows/generate-openapi.yml`. Similar to `.github/workflows/generate_json_schema.yml`, when changes to openapi schema (based on above tool), we make a new PR to update them at `.schema/openapi.json`

** Various fixes**
 - Some bugs from last minute change in: #3927 

## 🔨 Related Issues
 - #3927 introduced OpenAPI.

## 📄 Documentation Updates
 - OpenAPI docs to come.

## 🤔 Concerns
 - We should version `.schema/*.json` for each release.

### CODE REVIEW

File: crates/runtime/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+abc-sqlite.db-wal
```
Comment (Jeadie) [inline crates/runtime/.gitignore:1]: Produced by: https://github.com/spiceai/spiceai/blob/d21603282d9987dd1f121ff0de74fecf1fe03ab7/crates/runtime/src/dataaccelerator.rs#L481

File: crates/runtime/src/http/routes.rs [hunk 2]
```diff
@@ -236,65 +236,3 @@ fn cors_layer(cors_config: &CorsConfig) -> CorsLayer {
     cors.allow_methods([Method::GET, Method::POST, Method::PATCH])
         .allow_origin(allowed_origins)
 }
-
```
Comment (Jeadie) [inline crates/runtime/src/http/routes.rs:247]: This test is kind of pointless. No good way of programmatically doing it based on `axum::Router`.

File: crates/runtime/src/http/v1/chat.rs [hunk 3]
```diff
@@ -43,7 +43,6 @@ use tracing::{Instrument, Span};
     tag = "AI",
     request_body(
         description = "Create a chat completion request using a language model.",
-        required = true,
```
Comment (Jeadie) [inline crates/runtime/src/http/v1/chat.rs:46]: These aren't valid.

File: Cargo.toml [hunk 4]
```diff
@@ -24,7 +24,7 @@ members = [
     "tools/flightsubscriber/",
     "tools/otelpublisher/",
     "tools/spicepodschema/",
-    "tools/evalconverter",
+    "tools/evalconverter", "tools/spiceschema",
```
Comment (phillipleblanc) [inline Cargo.toml]: separate line?

File: tools/spiceschema/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        /// `--json` is an alias to format=json
+        #[arg(long, help = "Output in JSON format", conflicts_with_all = ["format", "yaml", "toml"])]
+        json: bool,
+
+        /// `--yaml` is an alias to format=yaml
+        #[arg(long, help = "Output in YAML format", conflicts_with_all = ["format", "json", "toml"])]
+        yaml: bool,
+
+        /// `--toml` is an alias to format=toml
+        #[arg(long, help = "Output in TOML format", conflicts_with_all = ["format", "json", "yaml"])]
```
Comment (phillipleblanc) [inline tools/spiceschema/src/main.rs]: Why is toml an output here if its not actually supported?

**Review by phillipleblanc — DISMISSED** [review]

**Review by sgrebnov — DISMISSED** [review]

File: crates/runtime/src/http/v1/packages.rs [hunk 6]
```diff
@@ -34,7 +34,7 @@ pub struct GeneratePackageRequest {
     pub params: HashMap<String, String>,
 }
 
-/// Generate a package from a GitHub source path.
+/// Zip Github source
```
Comment (phillipleblanc) [inline crates/runtime/src/http/v1/packages.rs]: Why this change? I think the previous comment is clearer

Comment (phillipleblanc) [reply crates/runtime/src/http/v1/packages.rs]: Or rename to `Generate Package` - the GitHub part is an implementation detail

**Review by phillipleblanc — DISMISSED** [review]

**Review by phillipleblanc — APPROVED** [review]
